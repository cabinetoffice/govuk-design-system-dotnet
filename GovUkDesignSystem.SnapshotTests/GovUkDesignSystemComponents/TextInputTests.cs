using System.Collections.Generic;
using System.Threading.Tasks;
using GovUkDesignSystem.GovUkDesignSystemComponents;
using GovUkDesignSystem.SnapshotTests.Helpers;
using Xunit;

namespace GovUkDesignSystem.SnapshotTests.GovUkDesignSystemComponents
{
    public class TextInputTests : SnapshotTestBase
    {
        private TextInputViewModel DefaultTextInputViewModel()
        {
            return new TextInputViewModel
            {
                Id = "test-id",
                Name = "test-name",
                Value = "test value",
                Type = "text",
                InputMode = "test-input-mode",
                DescribedBy = "test-described-by",
                Label = new LabelViewModel { Text = "test-label" },
                Hint = new HintViewModel { Text = "test-hint" },
                ErrorMessage = new ErrorMessageViewModel { Text = "test-error" },
                FormGroup = new FormGroupViewModel { Classes = "form-group-classes" },
                Autocomplete = "test autocomplete",
                Classes = "test-css-class",
                Pattern = "test-pattern",
                Spellcheck = true,
                Attributes = new Dictionary<string, string> { { "attr-name", "attr-value" } },
                TextInputAppendix = new TextInputAppendixViewModel { Text = "text-appendix" }
            };
        }

        [Fact]
        public async Task Render_AllValues()
        {
            // Act & Assert
            await VerifyPartial("TextInput", DefaultTextInputViewModel());
        }

        [Fact]
        public async Task Render_NoLabel()
        {
            // Arrange
            var viewModel = DefaultTextInputViewModel();
            viewModel.Label = null;

            // Act & Assert
            await VerifyPartial("TextInput", viewModel);
        }

        [Fact]
        public async Task Render_NoHint()
        {
            // Arrange
            var viewModel = DefaultTextInputViewModel();
            viewModel.Hint = null;

            // Act & Assert
            await VerifyPartial("TextInput", viewModel);
        }

        [Fact]
        public async Task Render_NoError()
        {
            // Arrange
            var viewModel = DefaultTextInputViewModel();
            viewModel.ErrorMessage = null;

            // Act & Assert
            await VerifyPartial("TextInput", viewModel);
        }

        [Fact]
        public async Task Render_NoFormGroup()
        {
            // Arrange
            var viewModel = DefaultTextInputViewModel();
            viewModel.FormGroup = null;

            // Act & Assert
            await VerifyPartial("TextInput", viewModel);
        }

        [Fact]
        public async Task Render_NoAppendix()
        {
            // Arrange
            var viewModel = DefaultTextInputViewModel();
            viewModel.TextInputAppendix = null;

            // Act & Assert
            await VerifyPartial("TextInput", viewModel);
        }

        [Fact]
        public async Task Render_NoDescribedBy()
        {
            // Arrange
            var viewModel = DefaultTextInputViewModel();
            viewModel.DescribedBy = null;
            viewModel.Hint = null;
            viewModel.ErrorMessage = null;

            // Act & Assert
            await VerifyPartial("TextInput", viewModel);
        }
    }
}
