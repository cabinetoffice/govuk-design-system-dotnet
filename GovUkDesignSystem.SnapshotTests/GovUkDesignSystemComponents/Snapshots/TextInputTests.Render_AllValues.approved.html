<div class="govuk-form-group form-group-classes govuk-form-group--error">
  <label class="govuk-label" for="test-id">

test-label            </label>
  <div class="govuk-hint" id="test-id-hint">

test-hint</div>
  <span class="govuk-error-message" id="test-id-error">
    <span class="govuk-visually-hidden">Error</span>

test-error</span>
  <input class="govuk-input test-css-class govuk-input--error" id="test-id" name="test-name" type="text" value="test value" aria-describedby="test-described-by test-id-hint test-id-error" autocomplete="test autocomplete" spellcheck="true" pattern="test-pattern" inputmode="test-input-mode" attr-name="attr-value" />


text-appendix</div>